src/Input.cpp

```diff
@@ -153,40 +154,168 @@ struct Gosu::Input::Impl
                 }
                 break;
             }
+            case SDL_JOYAXISMOTION: {
+                if (SDL_IsGameController(e->jaxis.which)) {
+                    // Ignore this event as joystick triggers are reported in range -1..1 instead of 0..1
+                    break;
+                }
+                else {
+                    store_axis_value(ButtonName::GP_LEFT_STICK_X_AXIS + e->jaxis.axis, (double)e->jaxis.value);
+
+                    if (int i = gamepad_slot_index(e->jaxis.which); i >= 0) {
+                        store_axis_value(ButtonName::GP_0_LEFT_STICK_X_AXIS + e->jaxis.axis + (6 * i), (double)e->jaxis.value);
+                    }
+                }
+                break;
+            }
+            case SDL_CONTROLLERAXISMOTION: {
+                store_axis_value(ButtonName::GP_LEFT_STICK_X_AXIS + e->caxis.axis, (double)e->caxis.value);
+
+                if (int i = gamepad_slot_index(e->caxis.which); i >= 0) {
+                    store_axis_value(ButtonName::GP_0_LEFT_STICK_X_AXIS + e->caxis.axis + (6 * i), (double)e->caxis.value);
+                }
+                break;
+            }
+            case SDL_JOYDEVICEADDED: {
```

Comment:
Hmm...not sure if this should happen in the event loop. Removed devices are enqueued, but added devices are handled immediately, this means the order of operations can swap. If through some glitch the device is removed and added in the same Input::update() call, it will temporarily exist in the device array twice.

I think the code would be easier to follow if the array modifications here would also happen in Input::update.